idaes/models_extra/power_generation/costing/power_plant_capcost.py

```diff
@@ -0,0 +1,1995 @@
+                    )
+                    * (
+                        pyunits.convert(scaled_param, ref_units)
+                        / (costing.ref_param[i] * ref_units)
+                    )
+                    ** costing.exp[i]
+                )
+
+        blk.bare_erected_cost_eq = Constraint(
+            cost_accounts, rule=bare_erected_cost_rule
+        )
+
+        # rule for calculating TPC
+        def total_plant_cost_rule(blk, i):
+            return blk.total_plant_cost[i] == blk.bare_erected_cost[i] * (
+                1 + blk.eng_fee[i] + blk.process_conting[i]
+            ) * (1 + blk.project_conting[i])
+
+        blk.total_plant_cost_eq = Constraint(cost_accounts, rule=total_plant_cost_rule)
+
+        # rule for sum of BEC
+        def BEC_sum_rule(blk):
+            return sum(blk.bare_erected_cost[i] for i in cost_accounts)
+
+        blk.bare_erected_cost_sum = Expression(rule=BEC_sum_rule)
+
+        # rule for sum of TPC
+        def TPC_sum_rule(blk):
+            return sum(blk.total_plant_cost[i] for i in cost_accounts)
+
+        blk.total_plant_cost_sum = Expression(rule=TPC_sum_rule)
+
+        # add variable and constraint scaling
+        for i in cost_accounts:
+            iscale.set_scaling_factor(blk.bare_erected_cost[i], 1)
+            iscale.set_scaling_factor(blk.total_plant_cost[i], 1)
+            iscale.constraint_scaling_transform(
+                blk.bare_erected_cost_eq[i], 1e-3, overwrite=False
+            )
+            iscale.constraint_scaling_transform(
+                blk.total_plant_cost_eq[i], 1, overwrite=False
+            )
+
+    # -----------------------------------------------------------------------------
+    # Supercritical CO2 Costing Library
+    # -----------------------------------------------------------------------------
+    def get_sCO2_unit_cost(
+        self,
+        equipment,
+        scaled_param,
+        temp_C=None,
+        n_equip=1,
+        CE_index_year="2018",
+        custom_accounts=None,
+    ):
+        """
+        Args:
+            self: pyomo Block where constraints will be made
+            unit_name: the name of the SCO2 equipment to cost
+            scaling_param: the scaling parameter (in appropriate units) for the
+                elected equipment
+            temp: the maximum temperature of the equipment. Not all types of
+                equipment use a temperature correction factor, so it is optional
+            n_equip: the number of pieces of equipment to cost
+
+        Cost is in M$
+        """
+        # check to see if a costing block already exists
+        if hasattr(self, "costing"):
+            raise AttributeError(
+                "{} already has an attribute costing. "
+                "Check that you are not calling get_costing"
+                " twice on the same model".format(self.name)
+            )
+
+        # create a costing Block
+        self.costing = Block()
```

Comment:
Since this is all being built in a `FlowsheetCostingBlock` I don't think we need the self.costing block anymore, everything can be stored in self.

Reply:
Agreed - we are already creating separate blocks to hold all of the costing information, so there is no need to add further blocks.